### DISCUSSION TITLE
Solution to disable 'body checks'

### DISCUSSION THREAD

**Discussion by KazzyJr** [agronholm/typeguard#450] [General]
Greetings!

I work in a team that uses **typeguard** as a library, specifically to decorate methods with the _typechecked_ decorator. We have recently upgraded from **2.x.x** to **4.x.x** and have hit a lot of snags in our testing pipeline, related to the fact that the old wrapping method was updated to a partial handling of the function. 
The main issue here is that **typeguard** checks the types within the body as well, and if a method argument typed as _str_ changes to an _int_ during a loop inside the function body, for example, **typeguard** raises an exception. This is a bit too strict for our development needs as we would like to still use Python's duck-typing down the way.

```python
@typechecked
def my_fun(arg1: str, arg2: int) -> list:
    while arg2 > 5:
        if arg1 == 'my_str':
            arg1 = arg2  # this is where it breaks
            arg2 -= 5
        else:
             arg2 -= 20
```

The question for you is: could a solution be implemented for the **typechecked** decorator that only limits the checking to the initial passed-on arguments (that enables the above example to pass)?

A few options from the top of mind below:
```python
@typechecked(body_check=False)
```
OR
```python
os.environ['TYPEGUARD_BODY_CHECK'] = 0
```

Thanks!

**Comment by agronholm** [comment]
Mypy is also going to complain about such errors, but I understand where you're coming from. I could accept a PR to this end where the flag is added as a configuration option (not env variable).

**Reply by agronholm** [reply]
Oh, right, so `typechecked()` will require a new parameter too.

**Reply by agronholm** [reply]
Any reason you're not using the import hook btw?

**Reply by KazzyJr** [reply]
We only decorate particular exposed methods, thus we need it targeted.
